src/pybind/mgr/dashboard/controllers/host.py

```diff
@@ -151,8 +155,25 @@ def merge_hosts_by_hostname(ceph_hosts, orch_hosts):
             }, orch_hosts_map[hostname]) for hostname in orch_hosts_map
     ]
     hosts.extend(orch_hosts_only)
+    for host in hosts:
+        host['service_instances'] = populate_service_instances(host['hostname'], host['services'])
```

Comment:
I think this can be improved, invocation to this function for each element is repeating lot of common stuff for each host.
For example, What about to call  "populate_service_instances" to obtain a dict like:
`srv_instances = {"hosta":[ "mgr", "ods1"], ... "hostb": ["mona", "crash", ...] ....}`

and after that loop over the host disct updating the values:
```
for host in hosts:
        host['service_instances'] = srv_instances(host['hostname'])
```

Reply:
@jmolmo : I think its necessary here to loop over each host all the time because in the end what I am trying to obtain is a dict like
```
{ 'mon': 3, 'mgr': 2, 'prometheus': 1}
```
and save it to the body of each of the host. that way in the front-end i can avoid the data processing which caused the issue in the first place.

Reply:
Ok. 
What I mean is that you are initializing and checking if available an "OrchClient.instance()" object for each host.. I do not know how "heavy" is that object ... but if we can avoid that passing the OrchClient.instance to the populate_service_instances function,, why do not to optimize a little bit the code?

Reply:
It's a singleton method (it only invokes the object creation for the first time and saves it inside the class for future calls), it shouldn't cause perf issues.